frontend/src/lib/components/TaxonomicFilter/infiniteListLogic.ts

```diff
@@ -296,17 +321,19 @@ export const infiniteListLogic = kea<infiniteListLogicType>([
             },
         ],
         localItems: [
-            (s) => [s.rawLocalItems, s.searchQuery, s.fuse],
-            (rawLocalItems, searchQuery, fuse): ListStorage => {
+            (s) => [s.rawLocalItems, s.searchQuery, s.swappedInQuery, s.fuse],
+            (rawLocalItems, searchQuery, swappedInQuery, fuse): ListStorage => {
                 if (rawLocalItems) {
-                    const filteredItems = searchQuery
-                        ? fuse.search(searchQuery).map((result) => result.item.item)
-                        : rawLocalItems
+                    const filteredItems =
+                        swappedInQuery || searchQuery
+                            ? fuse.search(swappedInQuery || searchQuery).map((result) => result.item.item)
+                            : rawLocalItems
 
                     return {
                         results: filteredItems,
                         count: filteredItems.length,
-                        searchQuery,
+                        searchQuery: swappedInQuery || searchQuery,
+                        originalQuery: swappedInQuery ? searchQuery : undefined,
```

Comment:
and so we only advertise there was an original query when we actually swapped something in